src/workerd/io/trace-stream.c++

```diff
@@ -1020,16 +1023,17 @@ kj::Promise<WorkerInterface::CustomEvent::Result> TailStreamCustomEventImpl::sen
   this->capFulfiller->fulfill(kj::mv(cap));
 
   try {
-    co_await sent.ignoreResult().exclusiveJoin(kj::mv(completionPaf.promise));
+    EventOutcome outcome = co_await sent.then([](auto resp) {
+      return resp.getResult();
+    }).exclusiveJoin(kj::mv(completionPaf.promise).then([]() { return EventOutcome::OK; }));
```

Comment:
Can we get a comment here on why completionPaf.promise should be OK?
Intuitively it feels more like "cancelled" to me but I can be convinced otherwise by a good comment 😄

Reply:
Addressed in the latest push, which is (finally) also informed by a proper understanding of why the stream cancellation warning above happens. With this approach the completion promise should only fulfill first if there's an actual cancellation, confirmed this locally with the two downstream PRs adding tests for stream cancellation and for the outcome event being reported properly.